### ISSUE TITLE
Avoid to move focus programmatically: Inserter menu and everywhere

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#685]
Splitting this out from #578 

For accessibility, moving focus programmatically should be done in very rare cases, for example when there's a single, very specific task to accomplish or in a modal, etc.

It should never be done in cases where moving focus will skip relevant content, as keyboard users would be forced to tab backwards to get to the skipped content and screen reader users won't even have a clue there is some content "before" the current focus.

In regard to the inserter menu, focus shouldn't be moved to the search field.

**Comment by mtias** [comment]
Closing this as the inserter has been modified a few times and is no longer problematic on this specific behaviour.
